staging/src/k8s.io/apimachinery/pkg/util/errors/errors.go

```diff
@@ -88,6 +95,22 @@ func (agg aggregate) Error() string {
 	return "[" + result + "]"
 }
 
+func (agg aggregate) As(target interface{}) bool {
+	var wasAssigned bool
+	agg.visit(func(err error) {
+		wasAssigned = wasAssigned || errors.As(err, target)
+	})
+	return wasAssigned
+}
+
+func (agg aggregate) Is(target error) bool {
+	var foundMatchingErr bool
+	agg.visit(func(err error) {
+		foundMatchingErr = foundMatchingErr || errors.Is(err, target)
```

Comment:
Add comment that indicates that this takes advantage of short-circuiting to avoid calling `As` after a single success.  Otherwise I fear it will be changed.

Reply:
It's common for traversal functions like this to have a boolean return to short-circuit. Since `visit` is private, it should be OK to add one.

Reply:
So just to confirm, you want me to adjust the signature of both `visit` itself and the passed-in func, because one out of two consumers in this package that will both need adjustment would then save three lines of code?

Reply:
No, the point would be avoiding the potentially large amount of unneeded function calls. Error checking hopefully isn't in a hot loop, so it probably doesn't matter here. (I should have marked this as just thinking out loud, not a request for a change, sorry)

Reply:
Done